solutions/gold/usaco-1019.mdx

```diff
@@ -0,0 +1,96 @@
+---
+id: usaco-1019
+source: USACO Gold 2020 February
+title: Delegation
+author: Chongtian Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_deleg_gold_feb20.html)
+
+## Explanation
+
+First note that if $K$ is not a divisor of the $N - 1$, then the edges can
+never be evenly partitioned. Thus, we only need to check divisors of $N - 1$.
+
+We can arbitrarily root the tree at node $1$. Let $\mathtt{toBeMerged[i]}$ store the length of the path passes through node $i$ and its length has not reached $K$ yet. Since only $1$ of such path exist, all of the other paths which source from the subtrees of $i$ must be evenly paired.
```

Comment:
here i'd give an example of what this array would look like
maybe take the sample input

Reply:
well the array is different for each k and making an image is kind of annoying sooo

Reply:
then just try for one k

Reply:
how do i add images
![image](https://github.com/cpinitiative/usaco-guide/assets/79723780/4eb3716f-c696-4d15-84ef-ba188ac73676)

Reply:
how to add images
![image](https://github.com/cpinitiative/usaco-guide/assets/79723780/6d12ddc2-9283-49f7-83b7-fa18c59e0ee8)

Reply:
how do i ad dimages
![image](https://github.com/cpinitiative/usaco-guide/assets/79723780/e1be5b97-a3ff-44c7-8da1-db74fb932b9d)

Reply:
perhaps take the text out of the image and put it in the editorial
as for how to add them, just look at other editorials with images, i think daniel zhu had some

Reply:
could yo put the graph into https://csacademy.com/app/graph_editor/ and take a screenshot

it looks kind of messy 👀

Reply:
fine, but my art skills are peak though